spa/src/components/portal/ContributionsList/ContributionDetail/ContributionDetail.tsx

```diff
@@ -30,6 +39,23 @@ export function ContributionDetail({ domAnchor, contributionId, contributorId }:
 
   useDetailAnchor(domAnchor ?? null, rootEl);
 
+  async function handlePaymentMethodUpdate(method: StripePaymentMethod) {
+    try {
+      await updateContribution({ provider_payment_method_id: method.id });
+      enqueueSnackbar(
+        'Your billing details have been successfully updated. Changes may not be reflected in portal immediately.',
+        {
+          persist: true,
+          content: (key: string, message: string) => (
+            <SystemNotification id={key} message={message} header="Billing Updated!" type="success" />
+          )
+        }
+      );
```

Comment:
@nrh-cklimas how about we add the custom success message as a input param to updateContribution?
As far as I could understand, the message is customized, but it will always show a success snackbar.
```suggestion
      await updateContribution({ provider_payment_method_id: method.id }, {
         "header": "Billing Updated!",
         "message": "Your billing details have been successfully updated. Changes may not be reflected in portal immediately.",
      });
```

Reply:
I like this idea a lot! I tweaked it a little to pass an abstract type of change and let the hook decide what message to show. b81f9334fb5eca91d1d11a840126a393940eaa4e